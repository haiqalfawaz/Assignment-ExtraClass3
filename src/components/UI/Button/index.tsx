import React from "react";
import { ButtonProps } from "./button.interface";

const Button = (props: ButtonProps) => {
  const { onClick, children } = props;

  const handleClick = () => {
    onClick();
  };
  return <button onClick={handleClick}>{children}</button>;
};

export default Button;
