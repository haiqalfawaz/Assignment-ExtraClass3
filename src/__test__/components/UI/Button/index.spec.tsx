import Button from "@/components/UI/Button";
import { render, fireEvent } from "@testing-library/react";

describe("Button", () => {
  it("should render button", () => {
    const handleClick = jest.fn();
    const { container } = render(<Button onClick={handleClick}>Click</Button>);
    expect(container).toMatchSnapshot();
  });
  it("should click button", () => {
    const handleClick = jest.fn();
    const { getByText } = render(<Button onClick={handleClick}>Click</Button>);
    const button = getByText("Click");
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
