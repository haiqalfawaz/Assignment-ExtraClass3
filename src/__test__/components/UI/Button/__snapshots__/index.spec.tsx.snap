// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button 1`] = `
<div>
  <button>
    Click
  </button>
</div>
`;
